crates/astria-core/src/sequencer/v1alpha1/block.rs

```diff
@@ -417,38 +463,53 @@ impl SequencerBlock {
             .try_into()
             .map_err(|e: Vec<_>| SequencerBlockError::incorrect_rollup_ids_root_length(e.len()))?;
 
-        let mut rollup_transactions = IndexMap::new();
+        let mut rollup_base_transactions = IndexMap::new();
         for elem in data_list {
             let raw_tx = raw::SignedTransaction::decode(&*elem)
                 .map_err(SequencerBlockError::signed_transaction_prootbof_decode)?;
             let signed_tx = SignedTransaction::try_from_raw(raw_tx)
-                .map_err(SequencerBlockError::raw_signed_transactin_conversion)?;
+                .map_err(SequencerBlockError::raw_signed_transaction_conversion)?;
             for action in signed_tx.into_unsigned().actions {
                 if let action::Action::Sequence(action::SequenceAction {
                     rollup_id,
                     data,
                     fee_asset_id: _,
                 }) = action
                 {
-                    let elem = rollup_transactions.entry(rollup_id).or_insert(vec![]);
+                    let elem = rollup_base_transactions.entry(rollup_id).or_insert(vec![]);
                     elem.push(data);
                 }
             }
         }
-        rollup_transactions.sort_unstable_keys();
+        rollup_base_transactions.sort_unstable_keys();
 
-        if rollup_transactions_root
-            != derive_merkle_tree_from_rollup_txs(&rollup_transactions).root()
-        {
+        // ensure the rollup IDs commitment matches the one calculated from the rollup data
+        if rollup_ids_root != merkle::Tree::from_leaves(rollup_base_transactions.keys()).root() {
+            return Err(SequencerBlockError::rollup_ids_root_does_not_match_reconstructed());
+        }
+
+        let rollup_transaction_tree = derive_merkle_tree_from_rollup_txs(&rollup_base_transactions);
+        if rollup_transactions_root != rollup_transaction_tree.root() {
             return Err(
                 SequencerBlockError::rollup_transactions_root_does_not_match_reconstructed(),
             );
         }
 
-        // ensure the rollup IDs commitment matches the one calculated from the rollup data
-        if rollup_ids_root != merkle::Tree::from_leaves(rollup_transactions.keys()).root() {
-            return Err(SequencerBlockError::rollup_ids_root_does_not_match_reconstructed());
+        let mut rollup_transactions = IndexMap::new();
+        for (i, (id, transactions)) in rollup_base_transactions.into_iter().enumerate() {
+            let proof = tree
```

Comment:
I believe the construction of these proofs are wrong. Reason:

1. `tree` is constructed from *all* the elements of `data`
2. `data[0]` and `data[1]` are `rollup_transactions_root` and `rollup_ids_root`, respectively.
3. this loop asks to construct proofs for leaves `0..len(rollup_txs)`, respectively, but that does not match the leaves of three.
4. Proof: try proving `rollup_base_transactions[0]` against the proof constructed for `i = 0` - you will not get the data hash.

Reply:
updated to use `rollup_transaction_tree` (the right tree) for constructing the proofs, will add tests